src/unix/lwt_preemptive.ml

```diff
@@ -18,121 +18,71 @@ open Lwt.Infix
    | Parameters                                                      |
    +-----------------------------------------------------------------+ *)
 
-(* Minimum number of preemptive threads: *)
-let min_threads : int ref = ref 0
-
-(* Maximum number of preemptive threads: *)
-let max_threads : int ref = ref 0
+(* Minimum number of domains: *)
+let min_channels : int ref = ref 0
+(* Maximum number of domains: *)
+let max_channels : int ref = ref 0
 
 (* Size of the waiting queue: *)
-let max_thread_queued = ref 1000
+let max_channel_queued = ref 1000
 
-let get_max_number_of_threads_queued _ =
-  !max_thread_queued
+let get_max_number_of_domains_queued _ =
+  !max_channel_queued
 
-let set_max_number_of_threads_queued n =
-  if n < 0 then invalid_arg "Lwt_preemptive.set_max_number_of_threads_queued";
-  max_thread_queued := n
+let set_max_number_of_domains_queued n =
+  if n < 0 then invalid_arg "Lwt_preemptive.set_max_number_of_domains_queued";
+  max_channel_queued := n
 
-(* The total number of preemptive threads currently running: *)
-let threads_count = ref 0
+(* The total number of domains currently running: *)
+let channels_count = ref 0
 
-(* +-----------------------------------------------------------------+
-   | Preemptive threads management                                   |
-   +-----------------------------------------------------------------+ *)
+  module C = Domainslib.Chan 
 
-module CELL :
-sig
-  type 'a t
-
-  val make : unit -> 'a t
-  val get : 'a t -> 'a
-  val set : 'a t -> 'a -> unit
-end =
-struct
-  type 'a t = {
-    m  : Mutex.t;
-    cv : Condition.t;
-    mutable cell : 'a option;
-  }
-
-  let make () = { m = Mutex.create (); cv = Condition.create (); cell = None }
-
-  let get t =
-    let rec await_value t =
-      match t.cell with
-      | None ->
-        Condition.wait t.cv t.m;
-        await_value t
-      | Some v ->
-        t.cell <- None;
-        Mutex.unlock t.m;
-        v
-    in
-    Mutex.lock t.m;
-    await_value t
-
-  let set t v =
-    Mutex.lock t.m;
-    t.cell <- Some v;
-    Mutex.unlock t.m;
-    Condition.signal t.cv
-end
-
-type thread = {
-  task_cell: (int * (unit -> unit)) CELL.t;
-  (* Channel used to communicate notification id and tasks to the
-     worker thread. *)
-
-  mutable thread : Thread.t;
-  (* The worker thread. *)
-
-  mutable reuse : bool;
-  (* Whether the thread must be re-added to the pool when the work is
-     done. *)
+
+type message = Do of (unit -> unit) | Quit 
+
+type chan = {
+  req : message C.t;
+  resp : unit C.t;
 }
 
-(* Pool of worker threads: *)
-let workers : thread Queue.t = Queue.create ()
+(* Pool of worker channels: *)
+let channels : chan Queue.t = Queue.create ()
 
```

Comment:
`worker` contains both thread and channel (in task_cell)

I think a similar structure should be present, although I'm not sure how to assign a type to a domain, is it a `'a Domain.t` ?